src/tests/test_magic.py

```diff
@@ -1987,3 +1988,29 @@ def test_accessing_previously_nonexisting_file(ip_empty, tmp_empty, capsys):
     ip_empty.run_cell("%sql SELECT * FROM 'data.csv' LIMIT 3")
     out, _ = capsys.readouterr()
     assert expected in out
+
+
+def test_warn_when_using_snippets_in_non_SELECT_command(ip_empty, capsys):
+    ip_empty.run_cell("%sql duckdb://")
+    ip_empty.run_cell("%sql create table languages (name VARCHAR, rating INTEGER)")
+    ip_empty.run_cell(
+        """%%sql
+INSERT INTO languages VALUES ('Python', 1), ('Java', 0), ('OCaml', 2)"""
+    )
+
+    ip_empty.run_cell(
+        """%%sql --save language_lt1
+select * from languages where rating < 1"""
+    )
+
+    with pytest.raises(UsageError) as _:
```

Comment:
looks like this is testing the create table case but you're not testing the case when the user has a CTE. please write more tests to check for any other relevant scenarios.

Reply:
Added more tests